packages/app/src/cli/services/dev/migrate-flow-extension.ts

```diff
@@ -0,0 +1,91 @@
+import {LocalSource, RemoteSource} from '../context/identifiers.js'
+import {IdentifiersExtensions} from '../../models/app/identifiers.js'
+import {getExtensionIds, LocalRemoteSource} from '../context/id-matching.js'
+import {
+  MigrateFlowExtensionMutation,
+  MigrateFlowExtensionSchema,
+  MigrateFlowExtensionVariables,
+} from '../../api/graphql/extension_migrate_flow_extension.js'
+import {partnersRequest} from '@shopify/cli-kit/node/api/partners'
+import {ensureAuthenticatedPartners} from '@shopify/cli-kit/node/session'
+import {AbortError} from '@shopify/cli-kit/node/error'
+
+export function getFlowExtensionsToMigrate(
+  localSources: LocalSource[],
+  remoteSources: RemoteSource[],
+  identifiers: IdentifiersExtensions,
+) {
+  const ids = getExtensionIds(localSources, identifiers)
+  const localExtensionTypesToMigrate = ['flow_action', 'flow_trigger']
+  const remoteExtensionTypesToMigrate = ['flow_action_definition', 'flow_trigger_definition']
+  const typesMap = new Map<string, string>([
+    ['flow_action', 'flow_action_definition'],
+    ['flow_trigger', 'flow_trigger_definition'],
+  ])
+
+  const local = localSources.filter((source) => localExtensionTypesToMigrate.includes(source.type))
+  const remote = remoteSources.filter((source) => remoteExtensionTypesToMigrate.includes(source.type))
+
+  const remoteSourcesMap = new Map<string, RemoteSource>()
+  remote.forEach((remoteSource) => {
+    remoteSourcesMap.set(remoteSource.uuid, remoteSource)
+    remoteSourcesMap.set(remoteSource.title, remoteSource)
+  })
+
+  return local.reduce<LocalRemoteSource[]>((accumulator, localSource) => {
+    const localSourceId = ids[localSource.configuration.name] ?? 'unknown'
+    const remoteSource = remoteSourcesMap.get(localSourceId) || remoteSourcesMap.get(localSource.configuration.name)
```

Comment:
Isn't the registration title equal to the name value in the toml?
Or what name/title property can we compare remote vs local?

Reply:
They are equal when first generated for the CLI-managed types (though can diverge after that). For existing extensions they can be different. The configuration's name (called "Action title"/"Trigger title" in the Partner Dashboard UI) is what is shown in the Flow editor so the linked PR changed the value put in the toml to the configuration's value (otherwise we would expose the partner's internal name in the Flow editor).

<img width="660" alt="image" src="https://github.com/Shopify/cli/assets/699739/fd53c1c4-6193-486d-9599-69b1dbae05a0">

Reply:
Updated to use the local extensions folder, as we know that it'd be unique 👌